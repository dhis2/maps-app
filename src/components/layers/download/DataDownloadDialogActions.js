import React from 'react';
import PropTypes from 'prop-types';
import { Button, ButtonStrip, CircularLoader } from '@dhis2/ui';
import i18n from '@dhis2/d2-i18n';
import styles from './styles/DataDownloadDialogActions.module.css';

export const DataDownloadDialogActions = ({
    downloading,
    onStartClick,
    onCancelClick,
}) => (
    <ButtonStrip end>
        <Button secondary onClick={onCancelClick} disabled={downloading}>
            {i18n.t('Cancel')}
        </Button>
        <Button primary onClick={onStartClick} disabled={downloading}>
            {i18n.t('Download')}
            {downloading && (
                <CircularLoader small className={styles.btnProgress} />
            )}
        </Button>
    </ButtonStrip>
);

DataDownloadDialogActions.propTypes = {
    downloading: PropTypes.bool.isRequired,
    onStartClick: PropTypes.func.isRequired,
    onCancelClick: PropTypes.func.isRequired,
};

export default DataDownloadDialogActions;
