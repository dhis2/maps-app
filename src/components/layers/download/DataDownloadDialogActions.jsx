import i18n from '@dhis2/d2-i18n'
import { Button, ButtonStrip, CircularLoader } from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import styles from './styles/DataDownloadDialogActions.module.css'

const DataDownloadDialogActions = ({
    downloading,
    onStartClick,
    onCancelClick,
}) => (
    <ButtonStrip end>
        <Button secondary onClick={onCancelClick} disabled={downloading}>
            {i18n.t('Cancel')}
        </Button>
        <Button
            name="download-button"
            primary
            onClick={onStartClick}
            disabled={downloading}
        >
            {i18n.t('Download')}
            {downloading && (
                <CircularLoader small className={styles.btnProgress} />
            )}
        </Button>
    </ButtonStrip>
)

DataDownloadDialogActions.propTypes = {
    downloading: PropTypes.bool.isRequired,
    onCancelClick: PropTypes.func.isRequired,
    onStartClick: PropTypes.func.isRequired,
}

export default DataDownloadDialogActions
