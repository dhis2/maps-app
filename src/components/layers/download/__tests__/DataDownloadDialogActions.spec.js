import React from 'react';
import { shallow } from 'enzyme';
import { DataDownloadDialogActions } from '../DataDownloadDialogActions';

describe('DataDownloadDialogActions', () => {
    const renderComponent = props =>
        shallow(
            <DataDownloadDialogActions
                classes={{}}
                downloading={false}
                onStartClick={() => null}
                onCancelClick={() => null}
                {...props}
            />
        );

    it('Should render two buttons and NO loading spinner', () => {
        const wrapper = renderComponent();
        expect(wrapper.find('WithStyles(ForwardRef(Button))').length).toBe(2);
        expect(
            wrapper.find('WithStyles(ForwardRef(CircularProgress))').length
        ).toBe(0);
    });

    it('Should disable buttons and show loading spinner when loading', () => {
        const wrapper = renderComponent({
            downloading: true,
        });
        expect(
            wrapper
                .find('WithStyles(ForwardRef(Button))')
                .at(0)
                .prop('disabled')
        ).toBe(true);
        expect(
            wrapper
                .find('WithStyles(ForwardRef(Button))')
                .at(1)
                .prop('disabled')
        ).toBe(true);
        expect(
            wrapper.find('WithStyles(ForwardRef(CircularProgress))').length
        ).toBe(1);
    });

    it('Should call onStartClick', () => {
        const fn = jest.fn();
        const wrapper = renderComponent({
            onStartClick: fn,
        });

        wrapper
            .find('WithStyles(ForwardRef(Button))[variant="contained"]')
            .simulate('click');
        expect(fn).toHaveBeenCalled();
    });

    it('Should call onCancel', () => {
        const fn = jest.fn();
        const wrapper = renderComponent({ onCancelClick: fn });

        wrapper
            .find('WithStyles(ForwardRef(Button))[variant="text"]')
            .simulate('click');
        expect(fn).toHaveBeenCalled();
    });
});
