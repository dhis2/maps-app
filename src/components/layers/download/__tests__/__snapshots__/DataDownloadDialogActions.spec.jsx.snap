// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataDownloadDialogActions Should render two buttons and NO loading spinner when not loading 1`] = `
<div>
  <div
    class="ui-ButtonStrip"
  >
    <div
      class="ui-Button"
    >
      Cancel
    </div>
    <div
      class="ui-Button"
    >
      Download
    </div>
  </div>
</div>
`;

exports[`DataDownloadDialogActions Should render two buttons and show loading spinner when loading 1`] = `
<div>
  <div
    class="ui-ButtonStrip"
  >
    <div
      class="ui-Button"
    >
      Cancel
    </div>
    <div
      class="ui-Button"
    >
      Download
      <div
        class="ui-CircularLoader"
      />
    </div>
  </div>
</div>
`;
